posthog/api/file_system/file_system.py

```diff
@@ -95,6 +95,11 @@ class FileSystemViewSet(TeamAndOrgViewSetMixin, viewsets.ModelViewSet):
     filter_backends = [filters.SearchFilter]
     pagination_class = FileSystemsLimitOffsetPagination
 
+    # The request has the team/environment in the URL, but want to filter by project not team.
+    param_derived_from_user_current_team = "project_id"
+    # This kludge is needed to avoid the default behavior of returning the project_id as the team_id
+    _skip_team_id_override_kludge = True
```

Comment:
That kludge only applies to endpoints nested with `project_id`, but in this PR you made the `file_system` endpoint `team_id`-based, so this _shouldn't_ be needed (unless I'm missing something)
https://github.com/PostHog/posthog/blob/fd70e5d86ec24d97e0021f69a16635f7552a27e5/posthog/api/__init__.py#L226-L227

Reply:
Without the kludge, I'd always get back the "main team" of the project (the one whose id is the same as the project's id), not the team that was actually in the URL